test/test_mavlink_protocol.cpp

```diff
@@ -122,83 +122,88 @@ static void play_stream(struct Context &ctx, const struct sockaddr_in &addr)
-        log_info("   Status: %s", info.status == 1 ? "streaming" : "not streaming");
-        log_info("   Resolution: %dx%d", info.resolution_h, info.resolution_v);
-        log_info("   URI: %s", info.uri);
+        reprint_streams(ctx);
+        play_stream(ctx, addr); 
+     
+        mavlink_message_t out_msg;
+        log_info("Select Mode:\n 0-Camera \n 1-Video ");
+       int option = read_int();
+       if(option==0){
+         mavlink_msg_command_long_pack(GCS_SYSID, MAV_COMP_ID_ALL, &out_msg,
+                             ctx.connected_camera_sysid, msg.compid,
+                             MAV_CMD_SET_CAMERA_MODE, 0, 0, 0, 0, 0, 0, 0, 0);
+        log_info("MAV_CMD_SET_CAMERA_MODE sent");
+        send_camera_msg(ctx, addr, out_msg);
+      }else if(option==1){
+        mavlink_msg_command_long_pack(GCS_SYSID, MAV_COMP_ID_ALL, &out_msg,
+                             ctx.connected_camera_sysid, msg.compid,
+                             MAV_CMD_SET_CAMERA_MODE, 0, 0, 1, 0, 0, 0, 0, 0);
+        log_info("MAV_CMD_SET_CAMERA_MODE sent");
+        send_camera_msg(ctx, addr, out_msg);
+      }else{
+        log_info("Invalid selection\n");}
 
-        if (info.status == 1) {
-            log_info("Video is already streaming to another client.");
-            exit(EXIT_FAILURE);
-        } else {
-            ret = snprintf(cmd, sizeof(cmd),
-                    "gst-launch-1.0 rtspsrc location=\"%s\" ! decodebin ! autovideosink sync=false",
-                    info.uri);
-            if (ret >= (int)sizeof(cmd) || system(cmd) <= 0) {
-                log_error("Unable to start video stream. Start it manually.");
-            }
-        }
-    } else if (msg.msgid == MAVLINK_MSG_ID_COMMAND_ACK) {
+        
+        mavlink_msg_command_long_pack(GCS_SYSID, MAV_COMP_ID_ALL, &out_msg,
+                                      ctx.connected_camera_sysid, msg.compid,
+                                      MAV_CMD_REQUEST_CAMERA_SETTINGS, 0, 1, 0, 0, 0, 0, 0, 0);       
+         log_info("MAV_CMD_REQUEST_CAMERA_SETTINGS sent");
+       send_camera_msg(ctx, addr, out_msg);
+      }else if(msg.msgid == MAVLINK_MSG_ID_CAMERA_SETTINGS){
+           log_info("Got MAVLINK_MSG_ID_CAMERA_SETTINGS");
+           mavlink_camera_settings_t settings;
+           mavlink_msg_camera_settings_decode(&msg, &settings);
+           switch(settings.mode_id){
+             case 0: log_info("image");break;
+             case 1: log_info("video");break;
+             case 2: log_info("image survey");break;
+          }
+
+
+               mavlink_message_t out_msg;
+                mavlink_msg_command_long_pack(GCS_SYSID, MAV_COMP_ID_ALL, &out_msg,
+                                      ctx.connected_camera_sysid, msg.compid,
+                                      MAV_CMD_IMAGE_START_CAPTURE, 0, 0, 1, 1, 0, 0, 0, 0);
+                  log_info("MAV_CMD_IMAGE_START_CAPTURE sent");
+                  send_camera_msg(ctx, addr, out_msg);
+
+                     
```

Comment:
I would suggest you to run git-clang-format before you submit. It would ease the review. :)